### PR TITLE
add additional tag support

### LINKED ISSUE #42: add a mechanism for users to specify tags
The Azure Subscription I am piloting in requires specific tags to be placed on all resources.
There doesnt appear to be a mechanism to allow me to specific my own tags to attach to resources when this is built.

### PR DESCRIPTION
**Pull request by ausfestivus**
This commit adds the code necessary to support additional tags.
Should be enough hopefully to fix #42

### CODE REVIEW

**Comment by ausfestivus** [comment]
Anyone know whats needed for this PR to get accepted?

**Comment by ausfestivus** [comment]
Hey @bnferguson can I redeem a cluepon?
Its not clear to me from the circleci fmt checks what the problem with the files are.

**Comment by ausfestivus** [comment]
Okay, formatting fixes applied.
Learnt stuff as a bonus. 🎉

**Comment by bnferguson** [comment]
@ausfestivus - Excellent! Yeah the CI is just looking for you to run `terraform fmt` locally which writes the changes. Probably could make that more obvious in the messaging.  

It may take me a minute to review this PR as I want to see if we can do pretty much the same change to all three modules as we're working to make the base interface between the three as similar as possible (making them easier to use), but in essence very much agree with this change of being able to add tags. 

Thanks for your patience and work here!

**Comment by ausfestivus** [comment]
No probs. 
I’m in the process of looking at some similar fmt goodness in a couple of pipelines in the current customers ADO environment. 

One thing with this PR that’s been at the back of my mind is to pull out the older embedded tagging stuff. This PR makes it redundant and the embedded tags don’t really add any value for a customer.

What’s your view?

I’m happy to hold this PR off till I can get the time to yoink the old tagging code out. I can probably squeeze some time in in the next week or two.

File: modules/configs/variables.tf [hunk 1]
```diff
@@ -95,6 +95,12 @@ variable "ca_bundle_url" {
   description = "URL to CA certificate file used for the internal `ptfe-proxy` used for outgoing connections"
 }
 
+variable "additional_tags" {
+  type        = "map"
+  description = "A map of additional tags to attach to all resources created."
+  default     = {}
+}
+
```
Comment (erindatkinson) [inline modules/configs/variables.tf]: I'm not sure if this is specifically needed. right now as the configs submodule doesn't create any Azure resources to be tagged.

File: main.tf [hunk 2]
```diff
@@ -63,6 +65,7 @@ module "configs" {
   weave_cidr           = "${var.weave_cidr}"
   repl_cidr            = "${var.repl_cidr}"
   release_sequence     = "${var.release_sequence}"
+  additional_tags      = "${var.additional_tags}"
```
Comment (erindatkinson) [inline main.tf]: same comment as with https://github.com/hashicorp/terraform-azurerm-terraform-enterprise/pull/48/files#r357280123

**Pull request merged**